pkg/plugin/plugin.go

```diff
@@ -164,13 +166,27 @@ func (p *Plugin) Health() error {
 		return err
 	}
 	if err != nil {
-		zap.L().Warn("health check fail", zap.Error(err))
+		zap.L().Warn("health check failed", zap.Error(err))
 	} else {
 		zap.L().Debug("health check success")
 	}
 	return err
 }
 
+// Checks the liveness of KMS API.
+// If the error returned from KMS is user-induced, the function returns nil.
+func (p *Plugin) Check() error {
```

Comment:
just naming suggestion, but wat about calling it Live instead of Check? (like if the /healthz endpoint calls Health to perform health check, then /livez endpoint calls Live to perform liveness check.)